app/views/records/edit_fields/_location_of_resources.html.erb

```diff
@@ -4,8 +4,12 @@
 <% else %>
 <% @clabel = nil? %>
 <% end %>
-
-<%= f.input :location_of_resources, label: @clabel, as: :nested_attribute_multiple, :fields => [:location_physical, :location_shelf_locator, :url], :label_html => { :class => 'multi-part-heading' } %>
+<% if not f.object.display_terms[:location_of_resources][:subfields].nil?%>
+<% @subfields = f.object.display_terms[:location_of_resources][:subfields].keys %>
```

Comment:
You also want the labels here to put them as labels on the fields. So you need the whole dictionary and use the values as labels: e.g. 
{:url=>"Url of the Resource", :location_physical=>"Physical Location"} 
URL field should have label "Url of the resource" isntead of "url"